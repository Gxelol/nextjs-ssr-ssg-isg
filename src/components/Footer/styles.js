import styled, { css } from 'styled-components';

import { Container as Text } from '../Text/styles';
import { Container as Section } from '../SectionContainer/styles';

export const Container = styled.footer`
  ${({ theme }) => css`
    text-align: center;
    border-top: 0.1rem solid ${theme.colors.primaryColor};

    a {
      color: inherit;
      text-decoration: none;
    }

    & ${Text} {
      font-size: ${theme.fonts.sizes.small};
    }

    & ${Section} {
      padding-top: 0;
      padding-bottom: 0;
    }
  `}
`;
