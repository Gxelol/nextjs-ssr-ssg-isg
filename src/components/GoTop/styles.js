import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    bottom: 2rem;
    right: 2rem;
    background: ${theme.colors.white};
    color: ${theme.colors.primaryColor};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    z-index: 6;
    opacity: 0.8;
  `}
`;
