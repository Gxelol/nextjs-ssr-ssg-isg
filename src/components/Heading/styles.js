import styled, { css } from 'styled-components';

const titleSize = {
  small: (theme) => css`
    font-size: ${theme.fonts.sizes.small};
  `,
  medium: (theme) => css`
    font-size: ${theme.fonts.sizes.medium};
    `,
  big: (theme) => css`
    font-size: ${theme.fonts.sizes.big};
    `,
  huge: (theme) => css`
    font-size: ${theme.fonts.sizes.huge};
    ${mediaFont(theme)}
    `,
};

const mediaFont = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.fonts.sizes.big};
  }
`;

const titleCase = (uppercase) => css`
text-transform: ${uppercase ? 'uppercase' : 'none'}
`;

export const Title = styled.h1`
  ${({ theme, darkColor, size, uppercase }) => css`
    color: ${darkColor ? theme.colors.primaryColor : theme.colors.white};
    ${titleSize[size](theme)};
    ${titleCase(uppercase)};
  `}
`;
