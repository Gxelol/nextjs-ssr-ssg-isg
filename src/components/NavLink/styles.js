import styled, { css } from 'styled-components';

export const Container = styled.nav`
  ${({ theme }) => css`
    display: flex;
    flex-flow: row wrap;
    text-transform: uppercase;

    @media ${theme.media.lteSmall} {
      flex-flow: column wrap;
      align-content: center;
    }
  `}
`;
