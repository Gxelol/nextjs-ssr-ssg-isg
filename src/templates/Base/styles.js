import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    padding-top: 3rem;

    @media ${theme.media.lteSmall} {
      padding-top: 0;
    }
  `}
`;
