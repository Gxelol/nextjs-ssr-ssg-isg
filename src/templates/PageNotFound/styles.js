import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    a {
      text-decoration: none;
      cursor: pointer;
      text-transform: uppercase;
    }
  `}
`;
