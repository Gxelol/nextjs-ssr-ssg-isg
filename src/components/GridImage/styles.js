import styled, { css } from 'styled-components';
import { Title as Heading } from '../Heading/styles';
import { Container as Text } from '../Text/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    > ${Text} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const GridContainer = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};

    @media ${theme.media.lteSmall} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    overflow: hidden;
  `}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 100%;
    transition: all 300ms ease-in-out;
    &:hover {
      transform: scale(1.2) rotate(10deg);
    }
  `}
`;
