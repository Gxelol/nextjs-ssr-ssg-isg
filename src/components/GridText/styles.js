import styled, { css } from 'styled-components';
import { Title as Heading } from '../Heading/styles';
import { Container as Text } from '../Text/styles';

export const Container = styled.section`
  ${({ theme }) => css`
    ${Text} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const GridContainer = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(28rem, 1fr));
    gap: ${theme.spacings.large};
    overflow: hidden;
    width: 100%;

    @media ${theme.media.lteSmall} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
  ${Heading} {
    position: relative;
    left: 5rem
  }

  ${Heading}::before {
    counter-increment: grid-counter;
    content: counter(grid-counter);
    position: absolute;
    font-size: 7rem;
    top: -3rem;
    left: -5rem;
    transform: rotate(5deg);
  }
  `}
`;
