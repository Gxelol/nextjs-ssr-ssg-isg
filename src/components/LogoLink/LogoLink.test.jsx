import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/renderTheme';
import { LogoLink } from '.';

describe('<LogoLink />', () => {
  it('should render text logo', () => {
    renderTheme(<LogoLink link="#target" text="Hello World" />);
    screen.getByRole('heading', { name: 'Hello World' }).tobe;
    expect(screen.getByRole('link', { name: 'Hello World' })).toHaveAttribute(
      'href',
      '#target',
    );
  });

  it('to match snapshot', () => {
    const { container } = renderTheme(
      <LogoLink link="#target" text="Hello World" srcImg="image.jpg" />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
