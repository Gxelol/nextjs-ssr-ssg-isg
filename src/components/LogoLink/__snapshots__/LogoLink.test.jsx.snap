// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoLink /> to match snapshot 1`] = `
<h1
  class="sc-gswNZR hhorMY"
>
  <a
    class="sc-bcXHqe cdkKrx"
    href="#target"
  >
    <img
      alt="Hello World"
      src="image.jpg"
    />
  </a>
</h1>
`;
